lib/src/runner/parse_metadata.dart

```diff
@@ -335,6 +365,62 @@ class _Parser {
     return constructorName;
   }
 
+  String _findConstructorNameFromMethod(
+      MethodInvocation constructor, String className) {
+    var target = constructor.target;
+    if (target != null) {
+      // target could be an import prefix or a different class. Assume that
+      // named constructor on a different class won't match the class name we
+      // are looking for.
+      if (constructor.methodName.name == className) return null;
+      // target is an optionally prefixed class, method is named constructor
+      String parsedName;
+      if (target is SimpleIdentifier) parsedName = target.name;
+      if (target is PrefixedIdentifier) parsedName = target.identifier.name;
```

Comment:
It would be helpful to show concrete examples of each case and which chunk of code matches them:

```dart
Timeout.factor()
prefix.Timeout()
prefix.Timeout.factor()
Timeout()
```

Reply:
Done